### PR TITLE
Improve toctree on coordinates landing page

### PR DESCRIPTION
**Pull request by dstansby**
Currently the table of contents for the coordinates landing page is hard to parse:

Screenshot 2023-03-22 at 14 23 55

This PR removes the "Examples" section headings, making for a slightly nicer and easier to read table of contents:
Screenshot 2023-03-22 at 14 24 40

### CODE REVIEW

**Review by eerovaher — CHANGES_REQUESTED** [review]
I would agree this is an improvement, and there are sub-packages that don't have any "Examples" section headings already, e.g. [`constants`](https://docs.astropy.org/en/stable/constants/index.html) or [`table`](https://docs.astropy.org/en/stable/table/index.html). Ideally we should try to make all the landing pages consistent by making analogous changes to all the sub-packages, but I'm not sure if it's best to do that in a single pull request or not.

I will point out that some of the source code for the `coordinates` landing page is in a separate file: https://github.com/astropy/astropy/blob/d3b5af50f6c1cfa14da0799c8835426d6db1a39e/docs/coordinates/index.rst#L540
That file should also be cleaned up: https://github.com/astropy/astropy/blob/d3b5af50f6c1cfa14da0799c8835426d6db1a39e/docs/coordinates/performance.inc.rst#L26-L27

**Comment by pllim** [comment]
@dstansby , are you able to address review comments by feature freeze on 2023-04-21? Thanks!

**Comment by dstansby** [comment]
Sorry, I won't have a chance to come back to this in the near future, feel free to close or someone else is welcome to push to my branch.

**Comment by pllim** [comment]
@eerovaher , are you interested to take over or should we just close this without merge?

**Review by mhvk — APPROVED** [review]
Yes, good (and fine to do it subpackage by subpackage)

**Comment by pllim** [comment]
Thanks, all!
